src/grpc_server.cc

```diff
@@ -3698,7 +3698,12 @@ ModelInferHandler::Process(InferHandler::State* state, bool rpc_ok)
     // has initiated... completion callback will transition to
     // COMPLETE. If error go immediately to COMPLETE.
     if (err != nullptr) {
-      LOG_VERBOSE(1) << "Infer failed: " << TRITONSERVER_ErrorMessage(err);
+      const char* request_id;
+      LOG_TRITONSERVER_ERROR(
+          TRITONSERVER_InferenceRequestIdString(irequest, &request_id),
+          "unable to retrieve request ID string");
+      LOG_VERBOSE(1) << request_id
+                     << "Infer failed: " << TRITONSERVER_ErrorMessage(err);
```

Comment:
I'm concerned `request_id` may spit out some garbage data or have other side effects being uninitialized here if RequestIdString call fails in any of these places:
```cpp
rmccormick@RMCCORMICK:~/triton/jira/3780-trt-reload/tritonserver$ cat test2.cpp 
#include <iostream>

int main()
{
  const char* request_id;
  std::cout << "request_id: [" << request_id << "] and some text" << std::endl;

  const char* request_id_init = "";
  std::cout << "request_id_init: [" << request_id_init << "] and some text" << std::endl;
  return 0;
}
rmccormick@RMCCORMICK:~/triton/jira/3780-trt-reload/tritonserver$ g++ test2.cpp 
rmccormick@RMCCORMICK:~/triton/jira/3780-trt-reload/tritonserver$ ./a.out 
request_id: [] and some text
request_id_init: [] and some text
```

Reply:
Good catch, and thanks for the code example. I initialized to an empty string to keep it simple.

The API call doesn't seem to have a way to fail currently (the error logging was to be extra safe), so there shouldn't be a case where request_id is overridden by garbage data.